### PR TITLE
blog post: tidync 2019-11-05

### PR DESCRIPTION
**Pull request by mdsumner**
Planned blog post on the tidync package for 5 November. 

Ping @stefaniebutland  - I still have a few waves of review to do but wanted to use the Hugo preview with the PR

### CODE REVIEW

**Comment by stefaniebutland** [comment]
@mdsumner tag me again here when you want me to review in detail.

Given the length, it will help to have 1-2 sentences near the start to lay out what the post will cover. If there's any way to have the title indicate what NetCDF data is, that would lure more uninitiated folk to read

**Comment by mdsumner** [comment]
Thanks @stefaniebutland please review! I'm having two smallish problems

- [ ] the figures don't showi up in the preview (I put them in roweb2/themes/ropensci/static/img/blog-images/2019-11-05-tidync/)
- [ ] pseudocode section doesn't number correctly  (all 1s).

**Comment by stefaniebutland** [comment]
1. `` works when I try it locally (i.e. remove `../../../roweb2/themes/ropensci/static` in front

preferred format is Hugo shortcode to link to figure like `{{< figure src="/img/blog-images/2019-11-05-tidync/oisst-data-1.png)" >}}` but not required

2. > pseudo-code section doesn't number correctly (all 1s).

Strange it's all 1s in preview but when I checkout your PR locally and do `hugo serve` it looks fine 
Screen Shot 2019-10-28 at 2 24 53 PM

**Comment by mdsumner** [comment]
I fixed the figures in the preview 👍

**Comment by stefaniebutland** [comment]
@maelle do you have a solution for item 2) here?: https://github.com/ropensci/roweb2/pull/531#issuecomment-547155281

>pseudo-code section doesn't number correctly (all 1s).

Strange it's all 1s in [preview](https://deploy-preview-531--ropensci.netlify.com/blog/2019/11/05/tidync/) but when I checkout the PR locally and do hugo serve it looks fine

**Comment by stefaniebutland** [comment]
Ready for me to review this @mdsumner ?

**Comment by mdsumner** [comment]
Oh yes please

**Comment by stefaniebutland** [comment]
will do today

File: content/blog/2019-11-05-tidync.md [hunk 1]
```diff
@@ -0,0 +1,1099 @@
...
+
+```R
+src_slc %>% hyper_array()
+
+src_slc %>% hyper_tibble()
+
+src_slc %>% hyper_tbl_cube()
+```
+
+There are various other packages for NetCDF in R, the main ones being [RNetCDF](https://CRAN.r-project.org/package=RNetCDF) and [ncdf4](https://CRAN.r-project.org/package=ncdf4). These are both *lower-level* tools than tidync - they are interfaces to the underlying NetCDF library, and tidync uses both to read information and data. The [raster](https://CRAN.r-project.org/package=raster), and [stars](https://CRAN.r-project.org/package=stars) packages provide quite different approaches and stars is more general than raster, but is similarly *higher-level* than tidync.
```
Comment (sckott) [inline content/blog/2019-11-05-tidync.md]: > underlying NetCDF library

link to the src?

Comment (sckott) [inline content/blog/2019-11-05-tidync.md]: > The raster, and stars packages

is that comma needed?

File: content/blog/2019-11-05-tidync.md [hunk 2]
```diff
@@ -0,0 +1,1099 @@
...
+
+To follow along with the code below requires all of the following packages, and we assume that recent versions are in use, particularly `ncmeta (>= 0.2.0)`, 
+`tidync (>= 0.2.2)`, and `tidyr (>= 1.0.0)`. 
+
+```R
+install.packages(c("ncmeta", "tidync", "maps", "stars", "ggplot2", "devtools", 
+                   "stars", "RNetCDF", "raster", "dplyr", "tidyr"))
+```
+
+The official website for tidync is https://docs.ropensci.org/tidync/ and the latest release can be found on [CRAN](https://CRAN.R-project.org/package=tidync).
```
Comment (sckott) [inline content/blog/2019-11-05-tidync.md]: maybe these links for where to find the pkg can be moved to the top of this section

File: content/blog/2019-11-05-tidync.md [hunk 3]
```diff
@@ -0,0 +1,1099 @@
...
+time_ex$time
+```
+
+```
+## [1] 1460
+```
+
+As mentioned, tidync considers time and its metadata a bit dangerous - it is often wrong, often inconsistent, there are time zone issues, sometimes *extra seconds* to account for and so we prefer to leave these interpretations to be validated manually  before automation. 
+
+Obtain the time units information and then use it to conver the raw value (`1460`) into a date-time understood by R.
```
Comment (sckott) [inline content/blog/2019-11-05-tidync.md]: convert

File: content/blog/2019-11-05-tidync.md (hunk 1 above)
Comment (mdsumner) [reply content/blog/2019-11-05-tidync.md]: will do

Comment (mdsumner) [reply content/blog/2019-11-05-tidync.md]: happy to remove

**Comment by mdsumner** [comment]
Thanks! All will be addressed soon.

**Comment by maelle** [comment]
@stefaniebutland what is your local Hugo version?

**Comment by maelle** [comment]
Not tried yet https://stackoverflow.com/questions/18088955/markdown-continue-numbered-list

**Comment by maelle** [comment]
Locally like Stef I get no issue (correct numbering) and run Hugo 0.58.3.

**Comment by maelle** [comment]
With a newer Hugo version the list is numbered correctly.
Preview: https://deploy-preview-539--ropensci.netlify.com/blog/2019/10/29/skimrv2/
PR: https://github.com/ropensci/roweb2/pull/539 (do not merge)

@stefaniebutland I guess we should upgrade Hugo, I can do that in master. #540  This won't fix this preview (Well I guess there are git processes for updating the fork etc. but maybe an overkill).

I also see the list looks weird (in Stef's screenshot, in the preview of my test linked above), there's a slight vertical gap between the number and the text but given future work on the website design we shouldn't try to fix that.

**Comment by stefaniebutland** [comment]
Thank you @maelle.

File: content/blog/2019-11-05-tidync.md (hunk 3 above)
Comment (mdsumner) [reply content/blog/2019-11-05-tidync.md]: @stefaniebutland 

> 1. Could you add a couple of sentences at the top to outline what you cover?

Considering this rewrite, should I include links to the main headers? 

```
In May 2019 version 0.2.0 of [tidync](https://docs.ropensci.org/tidync/) was approved by rOpenSci and accepted to CRAN. Here we provide a quick overview of the typical workflow with some pseudo-code for the main functions in tidync, this is enough to read if you just want to try out the package on your own data.  The tidync package is focussed on *efficient data extraction* for developing your own software, and this somewhat long post takes the time to explain the concepts in detail.  

There is a section about the [NetCDF](https://www.unidata.ucar.edu/software/netcdf/) data model itself. Then there is  a detailed illustration of a raster data set in R including some of challenges faced by R users, followed by sections on how tidync sees coordinates in NetCDF, and how we can *slice a dataset* and control the format of the output. We then discuss some limitations and future work, and then (most importantly) reflect on the rOpenSci process of package review. 
```

**Comment by mdsumner** [comment]
@stefaniebutland 

> 1. Could you add a couple of sentences at the top to outline what you cover?

Considering this rewrite, should I include links to the main headers? 

```
In May 2019 version 0.2.0 of [tidync](https://docs.ropensci.org/tidync/) was approved by rOpenSci and accepted to CRAN. Here we provide a quick overview of the typical workflow with some pseudo-code for the main functions in tidync, this is enough to read if you just want to try out the package on your own data.  The tidync package is focussed on *efficient data extraction* for developing your own software, and this somewhat long post takes the time to explain the concepts in detail.  

There is a section about the [NetCDF](https://www.unidata.ucar.edu/software/netcdf/) data model itself. Then there is  a detailed illustration of a raster data set in R including some of challenges faced by R users, followed by sections on how tidync sees coordinates in NetCDF, and how we can *slice a dataset* and control the format of the output. We then discuss some limitations and future work, and then (most importantly) reflect on the rOpenSci process of package review. 
```

File: content/blog/2019-11-05-tidync.md (hunk 1 above)
Comment (mdsumner) [reply content/blog/2019-11-05-tidync.md]: done

File: content/blog/2019-11-05-tidync.md (hunk 2 above)
Comment (mdsumner) [reply content/blog/2019-11-05-tidync.md]: done

File: content/blog/2019-11-05-tidync.md (hunk 3 above)
Comment (mdsumner) [reply content/blog/2019-11-05-tidync.md]: done

**Comment by mdsumner** [comment]
All done, thanks so much for the feedback @sckott and @stefaniebutland!  I know it's rather long, but this has been super helpful - I don't think I could have expressed that story as a vignette. 

@maelle I'll try to have a look at the numbering in the next days, thanks for the details!

**Comment by stefaniebutland** [comment]
@mdsumner (cc @maelle). The numbering issue has been fixed by Maëlle updating Hugo version (which is why numbers were ok when I checked locally) so will be fine when I publish.

**Comment by stefaniebutland** [comment]
The added intro info to outline the post, and the extra description of the oisst data set is great thank you.

Looks ready to publish. I'll likely do that in a way that suits your timezone (e.g. my Monday Nov 4 afternooon). Thanks so much @mdsumner !

**Comment by mdsumner** [comment]
Awesome!  All done, added links, fixed some minor typos, done my final read and check. 

Thank you @stefaniebutland @sckott and @maelle!

**Comment by mdsumner** [comment]
One last tweak for style: it's the royal-we through-out, but in the final acknowledgements I still use I/my.

**Comment by maelle** [comment]
I see the numbers look fine in the published version, great (the vertical misalignment doesn't look very good but as I mentioned earlier, more design work is planned on the website).

**Pull request merged**